### DISCUSSION TITLE
Trouble generating conversation title with Azure OpenAI

### DISCUSSION THREAD

**Discussion by rendybjunior** [danny-avila/LibreChat#4783] [Troubleshooting]
### What features would you like to see added?

Automatically infer what the conversation / session title should be.
The issue is there would be a lot of untitled "New Chat"  conversations where we don't know which one to go back to.
<img width="165" alt="image" src="https://github.com/user-attachments/assets/3e8a3ce0-1c7d-40a8-878b-38de5b377b49">

### More details

I have search issues using "auto name" and some variants but couldn't find anything similar

### Which components are impacted by your request?

UI

### Pictures

Example:
![image](https://github.com/user-attachments/assets/d432e606-673a-473b-a763-eb7c9ea8c578)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wsybok** [comment]
The issue seems to stem from some AI models being unable to summarize conversations. This could be due to two reasons: either the "summarize" parameter hasn't been enabled (set to "true"), or there's a formatting error when calling the summarization function. In my experience, Gemini fails to provide conversation titles, while the OpenAI model works correctly.

**Comment by rendybjunior** [comment]
turns out it works on chat using OpenAI, but doesn't work for chats using **Azure** OpenAI, any reason for that?

**Comment by ai-se-vikas** [comment]
titleConvo =true  and summarize= true 
In my case as well it doesn't show title of conversation. Can you please suggest any alternative method to handle this issue.

**Comment by danny-avila** [comment]
It works with azure, you need to set this in the `librechat.yaml` file:

```yaml
endpoints:
  azureOpenAI:
    titleModel: "gpt-4o-mini"
    titleConvo: true
```

More info: https://www.librechat.ai/docs/configuration/azure

**Reply by avimar** [reply]
Version: LibreChat v0.7.9-rc1 -
```
endpoints:
  azureOpenAI:
    plugins: false
    titleModel: "gpt-4.1-mini"
    titleConvo: true
    titleMethod: "completion"
```

I'm not getting any titles saved with azure either. Is it this error about `... .parse is not a function`?

```
2025-07-22T22:33:53.930Z debug: [saveConvo] api/app/clients/BaseClient.js - saveMessageToDatabase #saveConvo
2025-07-22T22:33:53.947Z debug: [AgentController] Request closed
2025-07-22T22:33:53.958Z error: [api/server/controllers/agents/client.js #titleConvo] Error this.client.chat.completions.parse is not a function
2025-07-22T22:33:53.959Z debug: [saveConvo] api/server/services/Endpoints/agents/title.js
2025-07-22T22:33:53.963Z debug: [AgentController] Title generation started
2025-07-22T22:33:53.964Z debug: [AgentController] Title generation completed
2025-07-22T22:33:53.964Z debug: [AgentController] Performing cleanup
2025-07-22T22:33:53.964Z debug: [AgentController] Cleaning up abort controller
2025-07-22T22:33:53.964Z debug: [AgentController] Cleanup completed
```

**Comment by LufinityLucas** [comment]
I'm also facing this issue recently. After upgrading to 0.7.9, it works but only after a refresh of the chat (either Ctrl + F5, or by enetering another chat and then going back to the one with the "New chat" title). I'm using Azure OpenAI as well, model 4.1-mini.

**Comment by GPTLeonard** [comment]
@danny-avila I'm experiencing the same issue with Azure OpenAI. It was working fine with `titleConvo: true` (that was always set to true), but ever since I set `"useResponsesApi: true"` to true, the `titleConvo` feature stopped working. Could this be related?

**Reply by Wartz** [reply]
I'm also experiencing this difficulty.

The `useResponsesAPI: true` setting seems to be necessary for Azure Foundry deployed models (and gpt-5).

**Reply by Wartz** [reply]
> @danny-avila I'm experiencing the same issue with Azure OpenAI. It was working fine with `titleConvo: true` (that was always set to true), but ever since I set `"useResponsesApi: true"` to true, the `titleConvo` feature stopped working. Could this be related?

So it turns out that **foundry** models, will title properly in librechat if you do not specify a baseURL;

When using the Azure Open AI method (`baseURL="{$RESOURCENAME}.cognitiveservices.azure.com/openai/v1/` and `"useResponsesAPI: true` in the model preset in modelSpecs, titling does NOT work.

I set addedEndpoints under model specs

```
modelSpecs:
    ....
    addedEndpoints:
    - azureOpenAI
```
